src/main/java/vn/triplet/controller/admin/HomeController.java

```diff
@@ -1,12 +1,33 @@
 package vn.triplet.controller.admin;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
 
-@Controller(value = "homeControllerOfAmin")
+@Controller
+@RequestMapping("/dashboard")
 public class HomeController {
-	@RequestMapping("/admin")
-	public String index() {
-		return "views/admin/home/index";
+	
+	@RequestMapping(value= {"","/"})
+	public String index(HttpServletRequest request,final RedirectAttributes redirectAttributes) 
+	{
+		HttpSession session = request.getSession();
+		if(session.getAttribute("roleUser")!=null) {
+			int role = (int) session.getAttribute("roleUser") ;
+			if(role==0) {
```

Comment:
cặp if -else từ line 20 - 25 có thể sửa cho  gọn như bên dưới nhé vì đã gặp return trong if, thì ko c ần else nữa, vì vào if return rồi thì code bên dưới ko chạy dc nữa mô. nên xóa else dc 

, ngoài ra, chị thấy có định nghĩa enum hay constant cho role á, em cso thể tận dụng thay vì settign 0 như thế này, vì đọc vô, ng ta ko bjk `0` là nghĩa admin hay uuser thường ...

message thì khai báo vào file  messages.properties nhé
`if (role==0) `
`return "views/admin/home/index";`

`redirectAttributes.addFlashAttribute("msg","You are not allow to access");`
`				return "redirect:/";`

Reply:
Dạ vâng, em đã sửa